webapp/src/components/NavigateRegionsOverlay.tsx

```diff
@@ -0,0 +1,347 @@
+    height: number;
+}
+
+type RegionId = "mainmenu" | "simulator" | "toolbox" | "editor" | "editortools" | "tutorial";
+
+interface Region {
+    id: RegionId;
+    ariaLabel: string;
+    shortcutKey: string;
+    getClassName?(projectView: IProjectView): string | undefined;
+    getBounds(projectView: IProjectView): DOMRect | undefined;
+    focus(projectView: IProjectView): void;
+}
+
+const getToolboxBounds = (projectView: IProjectView): DOMRect | undefined => {
+    return document.querySelector(`${projectView.isBlocksActive() ? ".blocklyToolbox" : ".monacoToolboxDiv"}`)?.getBoundingClientRect();
+}
+
+const isSimMini = () => !!document.querySelector(".miniSim");
+
+const regions: Region[] = [
+    {
+        id: "mainmenu",
+        ariaLabel: lf("Main menu"),
+        shortcutKey: "1",
+        getBounds() {
+            return document.querySelector("#mainmenu")?.getBoundingClientRect();
+        },
+        focus() {
+            findFirstFocusableDescendant(
+                document.querySelector("#mainmenu")!,
+                (e) =>
+                    // This isn't actually focusable and has the same behavior as home.
+                    e.classList.contains("brand") ||
+                    // The menuitem containing the editor toggle.
+                    !!Array.from(e.children).find(e => e.id === "editortoggle")
+            )?.focus();
+        }
+    },
+    {
+        id: "simulator",
+        ariaLabel: lf("Simulator"),
+        shortcutKey: "2",
+        getBounds(projectView: IProjectView) {
+            const element = isSimMini()
+                ? document.querySelector(".simPanel")
+                : projectView.state.collapseEditorTools ?
+                    document.querySelector("#computertogglesim")
+                    : document.querySelector("#editorSidebar")
+            const bounds = element?.getBoundingClientRect();
+            if (projectView.state.collapseEditorTools) {
+                // Custom presentation for the collapsed sim.
+                const isRtl = pxt.Util.isUserLanguageRtl();
+                const collapsedSimPadding = 30;
+                const copy = DOMRect.fromRect(bounds);
+                copy.y = bounds.top - collapsedSimPadding;
+                copy.width = 70;
+                copy.height = bounds.height + collapsedSimPadding * 2;
+                if (isRtl) {
+                    copy.x -= copy.width - bounds.width;
+                }
+                return copy;
+            }
+            return bounds;
+        },
+        focus(projectView: IProjectView) {
+            // Note that pxtsim.driver.focus() isn't the same as tabbing to the sim.
+            if (isSimMini()) {
+                projectView.setSimulatorFullScreen(true);
+            } else {
+                if (projectView.state.collapseEditorTools) {
+                    projectView.toggleSimulatorCollapse();
+                }
+                (document.querySelector("#boardview") as HTMLElement).focus();
```

Comment:
probably best to guard this in case boardview isn't defined.

Reply:
52dc00c1334d5ab7070a2f45d0aa4de069437064